### PR TITLE
Migrate IBusConfig to GSettings

### PR DESCRIPTION
**Pull request by fujiwarat**
IBus plans to deprecate IBusConfig and suggests to migrate to GSettings.
https://groups.google.com/forum/#!topic/ibus-devel/Mu1IoFX-bKE

https://github.com/choehwanjin/ibus-hangul/issues/51

### CODE REVIEW

File: src/engine.c [hunk 1]
```diff
@@ -248,104 +248,90 @@ ibus_hangul_engine_get_type (void)
...
 
-    value = ibus_config_get_value (config, "engine/hangul",
-                                         "switch-keys");
+    value = g_settings_get_value (settings_hangul, "switch-keys");
     if (value != NULL) {
         const gchar* str = g_variant_get_string (value, NULL);
         hotkey_list_set_from_string(&switch_keys, str);
-        g_variant_unref(value);
-    } else {
-	hotkey_list_append(&switch_keys, IBUS_Hangul, 0);
```
Comment (choehwanjin) [inline src/engine.c]: I would prefer handling the unpredictable error here.
Switch keys are quite important.

File: src/engine.c [hunk 2]
```diff
@@ -248,104 +248,90 @@ ibus_hangul_engine_get_type (void)
...
 
-    value = ibus_config_get_value (config, "engine/hangul",
-                                         "hanja-keys");
+    value = g_settings_get_value (settings_hangul, "hanja-keys");
     if (value != NULL) {
         const gchar* str = g_variant_get_string (value, NULL);
         hotkey_list_set_from_string(&hanja_keys, str);
-        g_variant_unref(value);
-    } else {
-	hotkey_list_append(&hanja_keys, IBUS_Hangul_Hanja, 0);
```
Comment (choehwanjin) [inline src/engine.c]: Same here.

**Comment by fujiwarat** [comment]
Updated the patch to follow your suggestion.
However I think it's better to call g_assert() instead of handling the errors but it's up to you.
Also you can modify my change after the patch is integrated.

**Comment by choehwanjin** [comment]
Thank you.

**Comment by fujiwarat** [comment]
Thank you!

**Pull request merged**